tests/uucore: build failure

```sh
cargo +stable build --workspace --tests
```

```log
error[E0425]: cannot find function `pipe` in module `pipes`
  --> src/uucore/src/lib/features/buf_copy.rs:73:54
   |
73 |         let (mut pipe_read, mut pipe_write) = pipes::pipe().unwrap();
   |                                                      ^^^^ not found in `pipes`
   |
note: found an item that was configured out
  --> src/uucore/src/lib/features/pipes.rs:22:8
   |
21 | #[cfg(any(target_os = "linux", target_os = "android"))]
   |          -------------------------------------------- the item is gated here
22 | pub fn pipe() -> std::io::Result<(File, File)> {
   |        ^^^^
help: consider importing one of these functions
   |
26 +     use std::io::pipe;
   |
26 +     use libc::pipe;
   |
26 +     use nix::unistd::pipe;
   |
26 +     use rustix::pipe::pipe;
   |
help: if you import `pipe`, refer to it directly
   |
73 -         let (mut pipe_read, mut pipe_write) = pipes::pipe().unwrap();
73 +         let (mut pipe_read, mut pipe_write) = pipe().unwrap();
   |

error[E0282]: type annotations needed
  --> src/uucore/src/lib/features/buf_copy.rs:76:13
   |
76 |             pipe_write.write_all(data).unwrap();
   |             ^^^^^^^^^^ cannot infer type
```

--- a/src/uucore/src/lib/features/pipes.rs
+++ b/src/uucore/src/lib/features/pipes.rs
@@ -11,18 +11,20 @@ use rustix::pipe::{SpliceFlags, fcntl_setpipe_size};
 use std::fs::File;
 #[cfg(any(target_os = "linux", target_os = "android"))]
 use std::os::fd::AsFd;
+#[cfg(any(target_os = "linux", target_os = "android"))]
 pub const MAX_ROOTLESS_PIPE_SIZE: usize = 1024 * 1024;
 
 /// A wrapper around [`rustix::pipe::pipe`] that ensures the pipe is cleaned up.
 ///
 /// Returns two `File` objects: everything written to the second can be read
 /// from the first.
-/// This is used only for resolving the limitation for splice: one of a input or output should be pipe
+/// used for resolving the limitation for splice: one of a input or output should be pipe
 #[inline]
-#[cfg(any(target_os = "linux", target_os = "android"))]
+#[cfg(any(target_os = "linux", target_os = "android", test))]
 pub fn pipe() -> std::io::Result<(File, File)> {
     let (read, write) = rustix::pipe::pipe()?;
     // improve performance for splice
+    #[cfg(any(target_os = "linux", target_os = "android"))]
     let _ = fcntl_setpipe_size(&read, MAX_ROOTLESS_PIPE_SIZE);
 
     Ok((File::from(read), File::from(write)))

